gel/_internal/_cache.py

```diff
@@ -8,56 +8,120 @@
 from typing import Any
 
 import json
+import pathlib
 
 from gel._internal import _atomic
 from gel._internal import _platform
 
 
-def save(key: str, data: bytes | str) -> bool:
-    cache_dir = _platform.cache_dir()
+_default_cache_dir = _platform.cache_dir()
 
+
+def save(
+    key: str,
+    data: bytes | str,
+    *,
+    cache_dir: pathlib.Path | None = None,
+    extra_key: str | None = None,
+) -> bool:
+    if cache_dir is None:
+        cache_dir = _default_cache_dir
     try:
         cache_dir.mkdir(parents=True, exist_ok=True)
     except OSError:
         return False
 
-    path = cache_dir / key
+    if extra_key is not None:
+        metadata = f"{key}.metadata"
+        try:
+            _atomic.atomic_write(cache_dir / metadata, extra_key)
+        except OSError:
+            return False
 
     try:
-        _atomic.atomic_write(path, data)
+        _atomic.atomic_write(cache_dir / key, data)
     except OSError:
         return False
 
     return True
 
 
-def save_json(key: str, data: Any) -> bool:
-    return save(key, json.dumps(data))
-
-
-def load_text(key: str) -> str | None:
-    path = _platform.cache_dir() / key
-
+def save_json(
+    key: str,
+    data: Any,
+    *,
+    cache_dir: pathlib.Path | None = None,
+    extra_key: str | None = None,
+) -> bool:
+    if cache_dir is None:
+        cache_dir = _default_cache_dir
+    return save(
+        key,
+        json.dumps(data),
+        cache_dir=cache_dir,
+        extra_key=extra_key,
+    )
+
+
+def _verify_cache(
+    key: str,
+    *,
+    cache_dir: pathlib.Path,
+    extra_key: str,
+) -> bool:
+    metadata = f"{key}.metadata"
     try:
-        return path.read_text(encoding="utf8")
+        value = (cache_dir / metadata).read_text(encoding="utf8")
+    except OSError:
+        return False
+    else:
+        return value == extra_key
+
+
+def load_text(
+    key: str,
+    *,
+    cache_dir: pathlib.Path | None = None,
+    extra_key: str | None = None,
+) -> str | None:
+    if cache_dir is None:
+        cache_dir = _default_cache_dir
+    if extra_key is not None:
+        _verify_cache(key, cache_dir=cache_dir, extra_key=extra_key)
```

Comment:
I don't think this does anything. `_verify_cache` returns whether the cache is legit but we are ignoring that.
Same with `load_bytes` and `load_json`